src/Client/ClientBase.cpp

```diff
@@ -1129,8 +1129,23 @@ void ClientBase::processInsertQuery(const String & query_to_execute, ASTPtr pars
     {
         /// If structure was received (thus, server has not thrown an exception),
         /// send our data with that structure.
+        bool change = false;
+        if (global_context->getInsertionTable().empty() && parsed_insert_query.table)
+        {
+            String table = parsed_insert_query.table->as<ASTIdentifier &>().shortName();
+            if (!table.empty())
+            {
+                change = true;
+                String database = parsed_insert_query.database ? parsed_insert_query.database->as<ASTIdentifier &>().shortName() : "";
+                global_context->setInsertionTable(StorageID(database, table));
+            }
+        }
```

Comment:
Let's make a separate method `setInsertionTable` in `ClientBase` for this logic and call it here. Also let's add a new method `hasInsertionTable` in context to check if we already have insertion table. Let's also skip changing it back to empty after sending data, it's not necessary I think.